README.md

```diff
@@ -1,8 +1,111 @@
+  token_url: "https://<tenant>.accounts400.ondemand.com/oauth2/token"
+```
+
+## Name reference between resources
+The Kyma CR, which creation is the trigger for the creation of the EventingAuth CR, uses the unqiue runtime ID of the managed Kyma runtime as name. This name is used as the name for the
+name for the EventingAuth CR and the IAS application. In this way, the eventing-auth CR and the IAS application can be assigned to the specific managed runtime.
+
+## Design decisions
+
+### Handling of Rate Limiting calling IAS API
+We didn't implement any rate limit handling, because the [Rate Limiting documentation of IAS](https://help.sap.com/docs/IDENTITY_AUTHENTICATION/6d6d63354d1242d185ab4830fc04feb1/e22ee47abf614565bcb29bb4ddbbf209.html) mentions the following: 
+> To ensure safe and stable environment, all requests have a limit of 50 concurrent requests per second. The requests are associated with the originating IP address, and not with the user making the requests.
+
+Currently, we do not expect to exceed this rate limit since a reconciliation can perform a maximum of 5 sequential requests.  
+There is also mention of a specific rate limit for SCIM endpoints, but we do not use these endpoints.
+
+### Caching of well-known token endpoint
+We read the known configuration of the IAS tenant that is used to create the applications to obtain the token endpoint. This token endpoint is then stored in the secret 
+on the managed runtime along with the client ID and the client secret.  
+The assumption is, that the token endpoint of the IAS tenant does not change without any notice of a breaking change.
+To reduce the number of requests when creating an application client secret and thus increase the stability of the reconciliation, it was decided to cache the 
+token endpoint on the first retrieval. The cached token endpoint is not invalidated during operator runtime, but is updated when the IAS credentials or tenant URL are changed.
+
+### Referencing IAS applications by name
+The IAS application is created with a name that matches the name of the EventingAuth CR. This name is the unique runtime ID of the cluster for which the IAS application is created.
+Since we do not want to store the IAS application ID in the secret stored on the managed runtime , we can read the IAS application only by its name.  
+During the creation of the application, existing applications with the same name are read. If an application with the same name exists, it is deleted, as we assume this is due to a failed reconciliation.
+If more than one application with the same name already exists, the reconciliation fails. The same behaviour occurs when reconciling the deletion of the EventingAuth CR.
+
+It was decided not to delete any of the existing applications in this case, as it is an unexpected condition that may have been caused by manual actions, and we may want to keep the applications to find the cause of the issue.
```

Comment:
Can we put something like this in a separate section with bullet points so that we can read them easily for our future improvements? For example:
### Future Improvements:
* Identify IAS Application with its UUID. Currently it is identified with its name.